src/Generator/Method/InterceptedMethod.php

```diff
@@ -43,4 +46,63 @@ public static function generateMethod(
 
         return $method;
     }
+
+    /**
+     * @return static
+     */
+    public static function fromReflectionWithoutBodyAndDocBlock(MethodReflection $reflectionMethod): self
+    {
+        /** @var static $method */
+        $method = static::copyMethodSignature($reflectionMethod);
+
+        $method->setInterface(false);
+        $method->setBody('');
+
+        if (\PHP_VERSION_ID < 80100) {
+            return $method;
+        }
+
+        /** @var callable(ReflectionMethod) : ReflectionMethod $getPrototype */
+        $getPrototype = \Closure::fromCallable([new ReflectionMethod(ReflectionMethod::class, 'getPrototype'), 'invoke']);
+
+        while (true) {
+            if ($reflectionMethod->hasTentativeReturnType()) {
+                $method->hasTentativeReturnType = true;
+                break;
+            }
+
+            if ($reflectionMethod->isAbstract()) {
+                break;
+            }
+
+            if (!$reflectionMethod->hasPrototype()) {
+                break;
+            }
+            try {
+                $reflectionMethod = $getPrototype($reflectionMethod);
+            } catch (ReflectionException $e) {
+                break;
+            }
+        }
+
+        return $method;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function copyMethodSignature(MethodReflection $reflectionMethod): \Laminas\Code\Generator\MethodGenerator
+    {
+        $method = parent::copyMethodSignature($reflectionMethod);
+
+        foreach ($reflectionMethod->getParameters() as $reflectionParameter) {
+            $method->setParameter(
+                $reflectionParameter->isPromoted()
+                    ? PromotedParameterGenerator::fromReflection($reflectionParameter)
+                    : InterceptedParameter::fromReflection($reflectionParameter)
+            );
+        }
+
+        return $method;
```

Comment:
maybe you can do this treatement at the end in `fromReflectionWithoutBodyAndDocBlock` so the result will be just 

```php
public static function fromReflectionWithoutBodyAndDocBlock(MethodReflection $reflectionMethod): self
{
	$method = parent::fromReflectionWithoutBodyAndDocBlock($reflectionMethod);
	foreach ($method->getParameters() as $reflectionParameter) {
            $method->setParameter(
                $reflectionParameter->isPromoted()
                    ? PromotedParameterGenerator::fromReflection($reflectionParameter)
                    : InterceptedParameter::fromReflection($reflectionParameter)
            );
        }
}